From: Jon Derrick <jonathan.derrick@intel.com>
Date: Wed, 12 Jul 2017 10:58:19 -0600
Subject: nvme: fabrics commands should use the fctype field for data direction
Patch-mainline: v4.13-rc3
Git-commit: 2fd4167fadd1360ab015e4f0e88e51843e49556c
References: bsc#1043805

Fabrics commands with opcode 0x7F use the fctype field to indicate data
direction.

Signed-off-by: Jon Derrick <jonathan.derrick@intel.com>
Reviewed-by: Sagi Grimberg <sai@grmberg.me>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Fixes: eb793e2c ("nvme.h: add NVMe over Fabrics definitions")
Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 include/linux/nvme.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/nvme.h b/include/linux/nvme.h
index f98d5834202b..15cf31893e11 100644
--- a/include/linux/nvme.h
+++ b/include/linux/nvme.h
@@ -915,7 +915,7 @@ static inline bool nvme_is_write(struct nvme_command *cmd)
 	 * Why can't we simply have a Fabrics In and Fabrics out command?
 	 */
 	if (unlikely(cmd->common.opcode == nvme_fabrics_command))
-		return cmd->fabrics.opcode & 1;
+		return cmd->fabrics.fctype & 1;
 	return cmd->common.opcode & 1;
 }
 
-- 
2.12.3

